### PR TITLE
Migrate docfx.E2ETest to .NET Core

### PR DESCRIPTION
**Pull request by qinezh**
@superyyrrzz @vwxyzh @hellosnow @ansyral @chenkennt

### CODE REVIEW

**Comment by vwxyzh** [comment]
:+1:

File: tools/Deployment/deploy.js [hunk 1]
```diff
@@ -455,13 +456,21 @@ let runSteps = function(promiseArray) {
...
 let clearReleaseStep = removePromiseFn(config.docfx.releaseFolder);
 let docfxBuildStep = execPromiseFn("build.cmd", ["Release", "PROD"], config.docfx.home);
-let e2eTestStep = execPromiseFn(config.msbuild, [config.docfx.e2eproj, "/p:Configuration=Release", "/t:Build"]);
 let genereateDocsStep = execPromiseFn(path.resolve(config.docfx.exe), ["docfx.json"], config.docfx.docFolder);
 let uploadDevMygetStep = uploadMygetPromiseFn(config.myget.exe, config.docfx.releaseFolder, config.myget.apiKey, config.myget.url.dev);
 let uploadMasterMygetStep = uploadMygetPromiseFn(config.myget.exe, config.docfx.releaseFolder, config.myget.apiKey, config.myget.url.master);
 
+let e2eTestStep = function() {
+  let stepsOrder = [
+    execPromiseFn(path.resolve(config.docfx.exe), ["docfx.json"], config.docfx.docfxSeedHome),
```
Comment (superyyrrzz) [inline tools/Deployment/deploy.js]: Also migrate PrepareForE2ETests.cmd, then remove RunE2ETests

**Comment by herohua** [comment]
:+1:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**